easy-ui-react/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,198 @@
+   */
+  isReadOnly?: boolean;
+
+  /**
+   * Whether the checkbox should be selected (controlled).
+   */
+  isSelected?: boolean;
+
+  /**
+   * Whether the checkbox is nested.
+   */
+  isNested?: boolean;
+
+  /**
+   * The name of the checkbox, used when submitting an HTML form.
+   */
+  name?: string;
+
+  /**
+   * Handler that is called when the checkbox's selection state changes.
+   */
+  onChange?: (isSelected: boolean) => void;
+
+  /**
+   * Size of the checkbox.
+   * @default md
+   */
+  size?: CheckboxSize;
+
+  /**
+   * Whether the input should display its "valid" or "invalid" visual styling
+   * @default valid
+   */
+  validationState?: ValidationState;
+
+  /**
+   * The value of the checkbox, used when submitting an HTML form.
+   */
+  value?: string;
+};
+
+/**
+ * A form element that enables a binary choice.
+ *
+ * @remarks
+ * Use a checkbox when users are required to make a binary choice.
+ *
+ * @example
+ * ```tsx
+ * <Checkbox>Checkbox item</Checkbox>
+ * ```
+ *
+ * @example
+ * _Default value:_
+ * ```tsx
+ * <Checkbox defaultSelected={true}>Checkbox item</Checkbox>
+ * ```
+ *
+ * @example
+ * _Controlled:_
+ * ```tsx
+ * <Checkbox
+ *   isSelected={isSelected}
+ *   onChange={(isSelected) => setIsSelected(isSelected)}
+ * >
+ *   Checkbox item
+ * </Checkbox>
+ * ```
+ */
+export function Checkbox(props: CheckboxProps) {
+  const {
+    children,
+    errorText,
+    isDisabled,
+    isIndeterminate,
+    isReadOnly,
+    isNested,
+    size = DEFAULT_SIZE,
+    validationState,
+  } = props;
+
+  const ref = React.useRef(null);
+
+  const state = useToggleState(props);
+  const { inputProps } = useCheckbox(props, state, ref);
+  const { isFocusVisible, focusProps } = useFocusRing();
+  const { isHovered, hoverProps } = useHover(props);
+
+  const isSelected = state.isSelected && !isIndeterminate;
+
+  const className = classNames(
+    styles.Checkbox,
+    isIndeterminate && styles.indeterminate,
+    isSelected && styles.selected,
+    isDisabled && styles.disabled,
+    isReadOnly && styles.readOnly,
+    isNested && styles.nested,
+    isFocusVisible && styles.focusVisible,
+    isHovered && styles.hovered,
+    styles[variationName("size", size)],
+    validationState === "invalid" && styles.invalid,
+  );
+
+  const textVariant =
+    size === "lg" ? "subtitle1" : isNested ? "body2" : "body1";
+  const textColor = isDisabled
+    ? "disabled"
+    : validationState === "invalid"
+    ? "danger"
+    : "primary";
+
+  if (size === "lg" && isNested) {
+    console.warn("isNested is incompatible with lg Checkbox");
+  }
+
+  return (
+    <span className={className} data-testid="root">
+      <label className={styles.label} {...hoverProps}>
+        <VisuallyHidden>
+          <input {...inputProps} {...focusProps} ref={ref} />
+        </VisuallyHidden>
+        <span className={styles.box}>
+          {(isIndeterminate || isSelected) && (
+            <span className={styles.mark}>
```

Comment:
**suggestion:** how do you feel about renaming the `mark` style to `check`? i think `check` is slightly more descriptive, but up to you

Reply:
good call